nifi-stateless/nifi-stateless-bundle/nifi-stateless-engine/src/main/java/org/apache/nifi/controller/reporting/LogComponentStatuses.java

```diff
@@ -0,0 +1,205 @@
+    private volatile long lastTriggerTime = System.currentTimeMillis();
+
+    public LogComponentStatuses(final FlowFileEventRepository flowFileEventRepository, final CounterRepository counterRepository, final FlowManager flowManager) {
+        this.flowFileEventRepository = flowFileEventRepository;
+        this.counterRepository = counterRepository;
+        this.flowManager = flowManager;
+
+        processorHeader = String.format(PROCESSOR_LINE_FORMAT, "Processor Name", "Processor ID", "Processor Type", "Bytes Read/sec", "Bytes Written/sec", "Tasks/sec", "Nanos/Task",
+            "Percent of Processing Time");
+        processorBorderLine = createLine(processorHeader);
+
+        counterHeader = String.format(COUNTER_LINE_FORMAT, "Counter Context", "Counter Name", "Counter Value", "Increase/sec");
+        counterBorderLine = createLine(counterHeader);
+    }
+
+    private String createLine(final String valueToUnderscore) {
+        final StringBuilder processorBorderBuilder = new StringBuilder(valueToUnderscore.length());
+        for (int i = 0; i < valueToUnderscore.length(); i++) {
+            processorBorderBuilder.append('-');
+        }
+        return processorBorderBuilder.toString();
+    }
+
+    @Override
+    public void run() {
+        try {
+            if (!logger.isInfoEnabled()) {
+                return;
+            }
+
+            logFlowFileEvents();
+            logCounters();
+        } catch (final Exception e) {
+            logger.error("Failed to log component statuses", e);
+        }
+    }
+
+    private void logFlowFileEvents() {
+        final long timestamp = System.currentTimeMillis();
+        final ProcessGroup rootGroup = flowManager.getRootGroup();
+        final List<ProcessorNode> allProcessors = rootGroup.findAllProcessors();
+
+        long totalNanos = 0L;
+        final List<ProcessorAndEvent> processorsAndEvents = new ArrayList<>();
+        for (final ProcessorNode processorNode : allProcessors) {
+            final FlowFileEvent flowFileEvent = flowFileEventRepository.reportTransferEvents(processorNode.getIdentifier(), timestamp);
+            if (flowFileEvent == null) {
+                continue;
+            }
+
+            processorsAndEvents.add(new ProcessorAndEvent(processorNode, flowFileEvent));
+            totalNanos += flowFileEvent.getProcessingNanoseconds();
+        }
+
+        final Comparator<ProcessorAndEvent> comparator = Comparator.comparing(procAndEvent -> procAndEvent.getEvent().getProcessingNanoseconds());
+        processorsAndEvents.sort(comparator.reversed());
+
+        final StringBuilder builder = new StringBuilder();
+        builder.append("Processor Statuses:\n");
+        builder.append(processorBorderLine);
+        builder.append("\n");
+        builder.append(processorHeader);
+        builder.append(processorBorderLine);
+        builder.append("\n");
+
+        final int secondsInEvent = 300; // TODO: Get this more intelligently
```

Comment:
This could be moved up to a private static final value for now, or the value `300` could be replaced with a static variable named something like `DEFAULT_SECONDS_IN_EVENT`.

Reply:
Yup good catch. Forgot to move that up. Will do.